semantic/tools/semantic2vef.cpp

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) 2021 Melown Technologies SE
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * *  Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * *  Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "service/cmdline.hpp"
+
+#include <cstdlib>
+#include <iostream>
+
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+
+#include "utility/buildsys.hpp"
+#include "utility/gccversion.hpp"
+#include "utility/streams.hpp"
+#include "utility/path.hpp"
+#include "utility/implicit-value.hpp"
+#include "utility/small_map.hpp"
+
+#include "vef/vef.hpp"
+
+#include "service/cmdline.hpp"
+
+#include "geometry/meshop.hpp"
+
+#include "semantic/io.hpp"
+#include "semantic/mesh.hpp"
+
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+
+namespace {
+
+const utility::small_map<semantic::Material, cv::Vec3b> materialColors(
+    { { semantic::Material::default_, cv::Vec3b(220, 220, 220) },
+      { semantic::Material::facade, cv::Vec3b(245, 245, 245) },
+      { semantic::Material::roof, cv::Vec3b(91, 114, 226) },
+      { semantic::Material::tree_trunk, cv::Vec3b(83, 53, 10) },
+      { semantic::Material::tree_crown_deciduous, cv::Vec3b(86, 137, 61) },
+      { semantic::Material::tree_crown_coniferous, cv::Vec3b(41, 73, 26) },
+      { semantic::Material::terrace, cv::Vec3b(71, 164, 233) } });
+
+class Semantic2Vef : public service::Cmdline
+{
+public:
+    Semantic2Vef()
+        : service::Cmdline("semantic2vef", BUILD_TARGET_VERSION)
+    {}
+
+private:
+    virtual void configuration(po::options_description &cmdline
+                               , po::options_description &config
+                               , po::positional_options_description &pd)
+        override;
```

Comment:
I would prefer to stick to the best practice and use only `override` keyword here.  Same for the other functions.

> CppCoreGuidelines:C.128: Virtual functions should specify exactly one of virtual, override, or final.

Reply:
Style copied from 
https://github.com/melowntech/libsemantic/blob/388bc4a638aaca62ef577b44c485e59cb38014b3/semantic/tools/semantic2gpkg.cpp#L31-L42
but ok, `virtual` does not have to be there.